assets/src/settings-page/supported-templates.js

```diff
@@ -1,49 +1,243 @@
+				onChange={
+					( newChecked ) => {
+						const newSupportedPostTypes = supportedPostTypes.filter( ( postType ) => postType !== postTypeObject.name );
+
+						if ( newChecked ) {
+							newSupportedPostTypes.push( postTypeObject.name );
+						}
+
+						updateOptions( { supported_post_types: newSupportedPostTypes } );
+					}
+				}
+			/>
+		</li>
+	);
+}
+PostTypeCheckbox.propTypes = {
+	postTypeObject: PropTypes.shape( {
+		label: PropTypes.string,
+		name: PropTypes.string,
+		supports_amp: PropTypes.bool,
+	} ).isRequired,
+};
+
+/**
+ * Container for the supported post type checkbox fieldset.
+ */
+function SupportedPostTypesFieldset() {
+	const { editedOptions, fetchingOptions } = useContext( Options );
+
+	const {
+		theme_support: themeSupport,
+		reader_theme: readerTheme,
+		supportable_post_types: supportablePostTypes,
+	} = editedOptions || {};
+
+	if ( fetchingOptions || ! supportablePostTypes ) {
+		return null;
+	}
+
+	const isLegacy = 'reader' === themeSupport && 'legacy' === readerTheme;
+
+	return (
+		<fieldset id="supported_post_types_fieldset">
+			{ ! isLegacy && (
+				<h4 className="title">
+					{ __( 'Content Types', 'amp' ) }
+				</h4>
+			) }
+			<p>
+				{ __( 'The following content types will be available as AMP:', 'amp' ) }
+			</p>
+			<ul>
+				{ supportablePostTypes.map( ( postTypeObject ) => {
+					return (
+						<PostTypeCheckbox
+							key={ `supportable-post-type-${ postTypeObject.name }` }
+							postTypeObject={ postTypeObject }
+						/>
+					);
+				} ) }
+			</ul>
+		</fieldset>
+	);
+}
+
+/**
+ * Get a list of the template IDs for the supported template and its descendants.
+ *
+ * @param {Object} supportableTemplate Supportable templates.
+ * @return {Array} Descendant template IDs, including the ID of the passed template.
+ */
+function getInclusiveDescendantTemplatesIds( supportableTemplate ) {
+	const templateIds = [ supportableTemplate.id ];
+	for ( const childSupportableTemplate of supportableTemplate.children ) {
+		templateIds.push( ...getInclusiveDescendantTemplatesIds( childSupportableTemplate ) );
+	}
+	return templateIds;
+}
 
-	const { reader_theme: readerTheme, theme_support: themeSupport } = editedOptions;
+/**
+ * List of checkboxes corresponding to supportable templates.
+ *
+ * @param {Object} props Component props.
+ * @param {Array} props.supportableTemplates Array of supportableTemplate objects.
+ */
+export function SupportedTemplatesCheckboxes( { supportableTemplates } ) {
+	const { editedOptions, updateOptions } = useContext( Options );
+
+	const { supported_templates: supportedTemplates } = editedOptions || {};
+
+	if ( ! supportableTemplates.length ) {
+		return null;
+	}
+
+	return (
+		<ul>
+			{ supportableTemplates.map( ( supportableTemplate ) => (
+				<li key={ supportableTemplate.id }>
+					<CheckboxControl
+						checked={ supportedTemplates.includes( supportableTemplate.id ) }
+						help={ supportableTemplate.description }
```

Comment:
Seems `description` is not a property here:

![image](https://user-images.githubusercontent.com/16200219/87608848-75430a80-c6f0-11ea-90f1-287d32b8714a.png)

Reply:
9105286f94624e36adf0687afd590c4aae25591c seems to be related.

Reply:
It will be populated if it is defined. We removed the only `description` that was being used in `AMP_Theme_Support::get_supportable_templates()`, so that is why they are all empty. A plugin can use the `amp_supportable_templates` filter to add their own templates and provide a `description`.

Reply:
Ah OK 👍.